### ISSUE TITLE
Installation and WaNo creation 

### ISSUE THREAD

**Issue by joergfunger** [KIT-Workflows/Fenics-Tensile#1]
I looked into simstack, but I'm not sure (based on the documentation below) how to add a WaNo that represents the fenics compression test. In addition, simstack crashes whenever I click on Jobs&Workflows and then e.g. on reload (or on one of the directories shown there). Finally, is there an option to test everything locally  (so e.g. just providing a docker file or a conda environment) that is executed on my machine?

**Comment by schaarj** [comment]
these are multiple issues. @timostrunk can probably help best with the crash but it might be an issue of an unsupported  combination of client/Server versions . 

Running jobs locally is not an official feature yet as the main concept of simstack is submitting jobs to a scheduler on remote HPC ressources. There is the option to use the experimental local backend, another thing for which @timostrunk is probably the best reference.

**Comment by timostrunk** [comment]
@joergfunger I agree, this sounds like a mismatch between client and server version. We have also observed this with ssh servers, which drop the connection after one request (Cisco ssh proxies to be precise) and older client versions.

Could you let me know about where and how you are executing this?

A local execution is possible with the Internal Queueing system. It's not officially supported and requires a local SimStack server install. We will work on builtin docker and support for conda environments in an upcoming project.

**Comment by joergfunger** [comment]
Currently, I have not installed any server yet (maybe that is the reason for the crash). It would be nice to test everything locally without requiring access to a server, because that complicates things. But assuming I have access to an HPC server, what would have to be installed there, the simstack server, or just conda (In case of the latter - how do you get the results back)? Where do I provide the conda environements for each WaNo? Do these environments have to be generated by myself (that is kind of tricky since the cluster is not connected to the web (conda create .. does not work).

**Comment by schaarj** [comment]
I would suggest to use one of our test servers for the time being. I can give you an account and help setting it up if interested. In order to run it on your resources you would need to get the Servercode from Nanomatch (@timostrunk ). In theory each WaNo should just source a config which takes care of making sure the actual code can be run on the resource. This can either be  a module/conda environment or manual configuration of paths.

**Comment by joergfunger** [comment]
How do you convince HPC system admins to install the server? On which HPC is that already installed? As for the configuration, where is that configuration defined?

**Comment by schaarj** [comment]
configuration is part of the Server install. which can be done in user space or a shared directory with write access, so no need to interact with Admins other than generation of a shared directory if needed. We are having a Server on our local HPC cluster for which we have simple user rights and a shared folder for all project partners.  Configuration is defined within that installation, and in most cases just a conda env that is sourced or a module that is loaded.

**Comment by schaarj** [comment]
We are also planning to include some further details on the usage of software in WaNos to the Simstack documentation. I briefly outlined the main idea in the following issue: https://github.com/simstack/Simstack-Documentation/issues/5

**Comment by timostrunk** [comment]
@joergfunger While most of the documentation to answer your questions does exist, it is not added to the official document store, which currently only explains client setup. I will touch it up and add it to the official read the docs, so you will have it within this week.

I would like to answer one question immediately though: SimStack is running only with user rights. It does not require any involvement from the HPC admin or external database. This is one of the reasons we actually started development of SimStack.
